Handle font licenses when editing theme metadata

matiasbenedetto opened the pull request:
## What
- Handle font licenses in the frontend 'edit theme metadata'.
- Adds a button to get theme fonts license text info updated.

## Why?
To avoid the need of collecting the font licesing information manually.
- This is part of the effort to re-introduce the font license theme automation (
Font licenses are no longer automated with new Core Font Library functionality #525).

## How
Calls the /font-families/ endpoint and process the data to create the fille the font credits text area.
Using a button to get the latest licenses has 2 benefits:
- Font assets metadata are loaded only when it is necessary.
- The theme author can add extra text to the one auto generated.

:warning: This PR depends on https://github.com/WordPress/create-block-theme/pull/648 being merged


## Screencast
This video features how the font licences can be updated for a theme.

[Screencast from 30-05-24 12:00:58.webm](https://github.com/WordPress/create-block-theme/assets/1310626/cffa1751-3491-48be-981b-a57955aa0160)

matiasbenedetto committed:
handle font licenses when editing theme metadata

matiasbenedetto committed:
move download file util

vcanales commented on src/resolvers.js:
```diff
@@ -85,3 +85,20 @@ export async function downloadExportedTheme() {
+		} );
+		return response.data;
+	} catch ( e ) {
+		// eslint-disable-next-line no-console
+		console.error( e );
```
Remove console statement

```suggestion
		
```

matiasbenedetto replied:
handled it here https://github.com/WordPress/create-block-theme/pull/649/commits/18839a3c37b6bff4cae725f81bd3b137c463c882

vcanales commented on src/utils/fonts.js:
```diff
@@ -0,0 +1,165 @@
+			licenseURL: nameTable.get( 14 ),
+		};
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error( 'Failed to get font license for font file.', error );
```
Remove console statement

```suggestion
		
```

matiasbenedetto replied:
handled it here https://github.com/WordPress/create-block-theme/pull/649/commits/18839a3c37b6bff4cae725f81bd3b137c463c882

vcanales commented on src/utils/fonts.js:
```diff
@@ -0,0 +1,165 @@
+		// Get the license from the font face url.
+		return await getFontFileLicenseFromUrl( faceUrl );
+	} catch ( error ) {
+		// eslint-disable-next-line no-console
+		console.error(
+			'Failed to get font license for font family: ',
+			fontFamily,
+			error
+		);
```
Remove console statement

```suggestion
		
```

vcanales reviewed (commented):
I see there are a few `console.error` calls; I think we should implement proper error management instead.

vcanales commented on src/editor-sidebar/metadata-editor-modal.js:
```diff
@@ -230,6 +239,45 @@ Plugin Description`,
+License
+Source`,
+						'create-block-theme'
+					) }
```
Translate each part individually instead.

```suggestion
					placeholder={ `${ __( 'Font Name', 'create-block-theme' ) }
${ __( 'Copyright', 'create-block-theme' ) }
${ __( 'License', 'create-block-theme' ) }
${ __( 'Source', 'create-block-theme' ) }` }
```

Also, perhaps modify to clarify how each part of the license should look like.

matiasbenedetto committed:
split string

Co-authored-by: Vicente Canales <1157901+vcanales@users.noreply.github.com>

matiasbenedetto committed:
remove console log

Co-authored-by: Vicente Canales <1157901+vcanales@users.noreply.github.com>

matiasbenedetto committed:
fix handle error in getting license

matiasbenedetto commented:
I removed the error handling where it was not needed and added it in the function where the license text is requested. If getting the license fails, the UI will throw an alert. 


Until merging https://github.com/WordPress/create-block-theme/pull/648 this PR will display this error:

[Screencast from 23-05-24 12:20:13.webm](https://github.com/WordPress/create-block-theme/assets/1310626/62c8b45b-ca19-42ca-b534-38262d59026c)

Why an alert? Because the snackbar is behind the modal component and is not visible. As that problem is not easy to fix (reference: https://github.com/WordPress/gutenberg/issues/52609) I think for now the alert is good enough. 

[Screencast from 23-05-24 12:02:53.webm](https://github.com/WordPress/create-block-theme/assets/1310626/d84c0247-6b84-4600-9676-7f81f70a5194)

vcanales reviewed (approved):
Looking good, although you'll probably have to update the branch with `trunk`, since there's a new eslint rule enforcing the specification of which rule is being disabled when eslint-disable comments are used.

matiasbenedetto committed:
Merge branch 'trunk' into add/handle-font-licenses-frontend
